components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -671,28 +674,70 @@ void AtmosphereDriver::create_fields()
   m_atm_logger->info("[EAMxx] create_fields ... done!");
 }
 
+void AtmosphereDriver::initialize_output_parameters () {
+  m_atm_logger->info("[EAMxx] initialize_output_parameters ...");
+  start_timer("EAMxx::init");
+  start_timer("EAMxx::initialize_output_parameters");
+
+  check_ad_status (s_comm_set | s_params_set);
+
+  auto& io_params = m_atm_params.sublist("Scorpio");
+
+  ekat::ParameterList checkpoint_params;
+  checkpoint_params.set("frequency_units",std::string("never"));
+  checkpoint_params.set("Frequency",-1);
+  if (io_params.isSublist("model_restart")) {
+    // Create model restart params
+    m_model_restart_output_params = io_params.sublist("model_restart");
+    m_model_restart_output_params.rename("model_restart_output_params");
+    m_model_restart_output_params.set<std::string>("Averaging Type","Instant");
+    m_model_restart_output_params.sublist("provenance") = m_atm_params.sublist("provenance");
+
+    // Store the "Output Control" pl of the model restart as the "Checkpoint Control" for all other output streams
+    checkpoint_params.set<std::string>("frequency_units",m_model_restart_output_params.sublist("output_control").get<std::string>("frequency_units"));
+    checkpoint_params.set("Frequency",m_model_restart_output_params.sublist("output_control").get<int>("Frequency"));
+  }
+
+  // Build one parameter list per output yaml file
+  using vos_t = std::vector<std::string>;
+  const auto& output_yaml_files = io_params.get<vos_t>("output_yaml_files",vos_t{});
+  int om_tally = 0;
+  for (const auto& fname : output_yaml_files) {
+    auto& params = m_output_managers_params.emplace_back();
+    ekat::parse_yaml_file(fname,params);
+    params.rename(ekat::split(fname,"/").back());
+    auto& checkpoint_pl = params.sublist("Checkpoint Control");
+    checkpoint_pl.set("frequency_units",checkpoint_params.get<std::string>("frequency_units"));
+    checkpoint_pl.set("Frequency",checkpoint_params.get<int>("Frequency"));
+
+    // Check if the filename prefix for this file has already been set.  If not, use the simulation casename.
+    if (not params.isParameter("filename_prefix")) {
+      params.set<std::string>("filename_prefix",m_casename+".scream.h"+std::to_string(om_tally++));
```

Comment:
I don't think we need the int after `.h`. Our naming convention is already taking care of differentiating two streams. If a user uses the same prefix for 2 streams that have the same output specs, well, then it's their fault. Though I suppose we could add a check in OutputManager, that ensures that when registering a file, we don't find out it was already registered...